zebra-state/src/service/finalized_state/disk_format/upgrade.rs

```diff
@@ -236,8 +241,8 @@ impl DbFormatChange {
         config: Config,
         network: Network,
         initial_tip_height: Option<Height>,
-        upgrade_db: ZebraDb,
-        should_cancel_format_change: Arc<AtomicBool>,
+        db: ZebraDb,
+        cancel_receiver: mpsc::Receiver<CancelFormatChange>,
```

Comment:
I thought we were using a single consumer for little overhead.

Reply:
Just a note in case this ever becomes a priority (but it's not at the moment, because we don't need concurrent upgrades yet):

`std::sync::mpsc::Receiver` takes a shared reference, so it can be called on `Arc<Receiver>`:
https://doc.rust-lang.org/std/sync/mpsc/struct.Receiver.html#method.try_recv

(This is different from `tokio::sync::mpsc::Receiver`, which requires an exclusive reference.)

However, if we were using an `mpsc` channel from multiple threads, only the first thread would get the cancel message. The second thread would keep running until the sender was dropped. (Which doesn't depend on the update thread finishing, so that's ok.)

A correct cloneable implementation would be `matches!(tokio::sync::watch::Receiver::has_changed(), Ok(true) | Err(_))`:
https://docs.rs/tokio/latest/tokio/sync/watch/struct.Receiver.html#method.has_changed

There might also be other ways of doing it.